### PR TITLE
[GEP-7] Enhance migration, fix BackupEntry annotation bug

### PR DESCRIPTION
**Pull request by swilen-iwanow**
**How to categorize this PR?**

/area control-plane-migration
/kind enhancement
/kind bug
/priority normal

**What this PR does / why we need it**:
Enhances the migration flow by adding steps for: 
- Kubernetes API Server scale-up, in case of hibernated Shoot
- Annotating BackupEntry for migration (there was a bug that was annotating every BackupEntry with migrate in the Seed)

**Which issue(s) this PR fixes**:
Fixes partially #1631  
**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/registry/core/shoot/strategy.go [hunk 1]
```diff
@@ -172,6 +168,11 @@ func (shootStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 	newShoot := obj.(*core.Shoot)
 	oldShoot := old.(*core.Shoot)
 	newShoot.Spec = oldShoot.Spec
+
+	if lastOperation := newShoot.Status.LastOperation; lastOperation != nil && lastOperation.Type == core.LastOperationTypeRestore &&
```
Comment (rfranzke) [inline pkg/registry/core/shoot/strategy.go:172]: Why was that moved out of the `mustIncreaseGeneration` function?

**Review by rfranzke — COMMENTED** [review]
Generally looks good, just one question.

/invite @plkokanov @vlvasilev @stoyanr

File: pkg/registry/core/shoot/strategy.go (hunk 1 above)
Comment (swilen-iwanow) [reply pkg/registry/core/shoot/strategy.go:172]: In the previous function shoot.oldStatus is set to the shoot.newStatus meaning lastOperation was old not Restore=Pending, so it needed one more update in order to increase the generation.

Comment (rfranzke) [reply pkg/registry/core/shoot/strategy.go:172]: I see, thank you!

**Comment by swilen-iwanow** [comment]
@vlvasilev could you drop a review, mainly for the worker actuator : )

**Review by rfranzke — DISMISSED** [review]
/lgtm
/needs second-opinion

File: pkg/operation/botanist/controlplane.go [hunk 2]
```diff
@@ -321,12 +346,13 @@ func (b *Botanist) ScaleGardenerResourceManagerToOne(ctx context.Context) error
 	return kubernetes.ScaleDeployment(ctx, b.K8sSeedClient.Client(), kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameGardenerResourceManager), 1)
 }
 
-// PrepareControlPlaneDeploymentsForMigration scales the kube-apiserver, kube-controller-manager and gardener-resource-manager to zero and deletes the hvpa for the kube-apiserver
-func (b *Botanist) PrepareControlPlaneDeploymentsForMigration(ctx context.Context) error {
+// PrepareKubeAPIServerForMigration deletes the kube-apiserverand deletes its hvpa
```
Comment (vlvasilev) [inline pkg/operation/botanist/controlplane.go]:
```suggestion
// PrepareKubeAPIServerForMigration deletes the kube-apiserver and deletes its hvpa
```

File: pkg/operation/botanist/controlplane.go [hunk 3]
```diff
@@ -321,12 +346,13 @@ func (b *Botanist) ScaleGardenerResourceManagerToOne(ctx context.Context) error
...
+func (b *Botanist) PrepareKubeAPIServerForMigration(ctx context.Context) error {
```
Comment (vlvasilev) [inline pkg/operation/botanist/controlplane.go:350]: Maybe it will be better to put the deletion of the hvpa in separated function as the deletion of the kupe-apiserver deployment

Comment (swilen-iwanow) [reply pkg/operation/botanist/controlplane.go:350]: I am not sure what will be the actual benefit except that it could be reused in future. But I think in most cases deleting the KubeAPI server will require cleaning the hvpa as well, so you will need to add both steps again? WDYT?

**Review by vlvasilev — APPROVED** [review]
/LGTM

**Review by rfranzke — APPROVED** [review]
